dvc/scm/tree.py

```diff
@@ -58,20 +61,50 @@ def isfile(self, path):
         """Test whether a path is a regular file"""
         return os.path.isfile(path)
 
-    def walk(self, top, topdown=True, dvcignore=None):
+    def walk(self, top, topdown=True, dvcignore=None, **kwargs):
         """Directory tree generator.
 
         See `os.walk` for the docs. Differences:
         - no support for symlinks
         - it could raise exceptions, there is no onerror argument
         """
 
-        assert dvcignore
-
         def onerror(e):
             raise e
 
         for root, dirs, files in dvc_walk(
             os.path.abspath(top), dvcignore, topdown=topdown, onerror=onerror
         ):
```

Comment:
Still using dvc_walk with dvcignore, which you probably shouldn't since you have CleanTree now 🙂 Also, maybe we could get rid of dvc_walk at all, since we now have CleanTree?